packages/datadog-plugin-http2/src/client.js

```diff
@@ -20,6 +22,75 @@ const HTTP2_HEADER_PATH = ':path'
 const HTTP2_HEADER_STATUS = ':status'
 const HTTP2_METHOD_GET = 'GET'
 
+class Http2ClientPlugin extends Plugin {
+  static get name () {
+    return 'http2'
+  }
+
+  constructor (...args) {
+    super(...args)
+
+    this.addSub('apm:http2:client:request:start', ({ authority, options, headers = {} }) => {
+      const sessionDetails = extractSessionDetails(authority, options)
+      const path = headers[HTTP2_HEADER_PATH] || '/'
+      const pathname = path.split(/[?#]/)[0]
+      const method = headers[HTTP2_HEADER_METHOD] || HTTP2_METHOD_GET
+      const uri = `${sessionDetails.protocol}//${sessionDetails.host}:${sessionDetails.port}${pathname}`
+
+      const store = storage.getStore()
+      const childOf = store ? store.span : store
+      const span = this.tracer.startSpan('http.request', {
+        childOf,
+        tags: {
+          [SPAN_KIND]: CLIENT,
+          'service.name': getServiceName(this.tracer, this.config, sessionDetails),
+          'resource.name': method,
+          'span.type': 'http',
+          'http.method': method,
+          'http.url': uri
+        }
+      })
+
+      addHeaderTags(span, headers, HTTP_REQUEST_HEADERS, this.config)
+
+      if (!hasAmazonSignature(headers, path)) {
+        this.tracer.inject(span, HTTP_HEADERS, headers)
+      }
+
+      analyticsSampler.sample(span, this.config.measured)
+
+      this.enter(span, store)
+    })
+
+    this.addSub('apm:http2:client:response', (headers) => {
+      const span = storage.getStore().span
+      const status = headers && headers[HTTP2_HEADER_STATUS]
+
+      span.setTag(HTTP_STATUS_CODE, status)
+
+      if (!this.config.validateStatus(status)) {
+        this.addError()
+      }
+
+      addHeaderTags(span, headers, HTTP_RESPONSE_HEADERS, this.config)
+
+      span.finish()
+    })
+
+    this.addSub('apm:http2:client:request:finish', () => {
+      const span = storage.getStore().span
+
+      span.finish()
```

Comment:
You're calling span.finish() in both this and the response event. If this is correct, why/how is this correct?

Reply:
It was not correct, I removed the redundant call.